third_party/move/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -747,13 +747,28 @@ impl CompiledPackage {
             if skip_attribute_checks {
                 flags = flags.set_skip_attribute_checks(true)
             }
-            let model = run_model_builder_with_options_and_compilation_flags(
-                vec![sources_package_paths],
-                deps_package_paths.into_iter().map(|(p, _)| p).collect_vec(),
-                ModelBuilderOptions::default(),
-                flags,
-                &known_attributes,
-            )?;
+            let model = if resolution_graph.build_options.generate_docs
+                || resolution_graph.build_options.generate_abis
+            {
+                run_model_builder_with_options_and_compilation_flags(
+                    vec![sources_package_paths],
+                    deps_package_paths.into_iter().map(|(p, _)| p).collect_vec(),
+                    ModelBuilderOptions::default(),
+                    flags,
+                    &known_attributes,
+                )?
+            } else if let Some(env) = optional_global_env {
+                // Use V2 generated model if not for docgen or abigen
+                env
+            } else {
+                run_model_builder_with_options_and_compilation_flags(
```

Comment:
Looks like this code is identical to the `if` body. Can we please refactor the code to remove code dup (eg., check if `optional_global_env` is `None` in the if conditional, or similar).

Reply:
done